### ISSUE TITLE
block.json: Expose supports > layout via block attributes and allow it to be overwritten

### ISSUE THREAD

**Issue by danieldudzic** [WordPress/gutenberg#53094]
## What problem does this address?
In my custom block, I'm trying to control the block layout by utilizing the core blocks, without introducing unnecessary complexity/nesting.

For now, I am using a **Group** block wrapper, which I am changing based on the block settings selected.

![255547514-33097810-554d-4f97-b21a-297d4a2e33ce](https://github.com/WordPress/gutenberg/assets/905781/7cd148d0-643c-4333-93f9-0df1244111a2)

I wish I could utilize:

```
{
	"supports": {
		"layout": {
			"default": {
				"type": "flex",
				"flexWrap": "nowrap"
			}
		}
	}
}

```

and then just update the `layout` via `edit.tsx` in the same manner I'm doing with the **Group** block attributes:

```
const block = select( 'core/block-editor' ).getBlock( clientId );
	block?.innerBlocks.forEach( ( innerBlock ) => {
		if ( innerBlock.name === 'core/group' ) {
			const updatedBlock = {
				...innerBlock,
				attributes: {
					...innerBlock.attributes,
					layout: setGroupAttributes( attributes.thumbnailsPosition ),
				},
			};
			dispatch( 'core/block-editor' ).updateBlock(
				innerBlock.clientId,
				updatedBlock
			);
		}
	} );


```
This would enable the same functionality, without the unnecessary **Group** block wrapper.

## What is your proposed solution?
Expose the `"supports": { "layout": {} }` via block attributes proving a way to update them.

**Comment by t-hamano** [comment]
Hi @danieldudzic,

For example, what about adding your own layout attributes to the custom block itself, instead of using a group block?

If that attribute is changed from the block sidebar, you could rearrange and reorder the inner block at the same time the attribute is changed.

Perhaps code something like this:

```javascript
const onChangeLayout = ( newLayout ) => {
	if ( newLayout === 'thumbnail-right' ) {
		// Relocate child blocks...
	} else if ( newLayout === 'thumbnail-left' ) {
		// Relocate child blocks...
	} else if ( newLayout === 'no-thumbnail' ) {
		// Relocate child blocks...
	}
	setAttributes( { layout: newLayout } );
};
```

Would this be helpful?

**Comment by danieldudzic** [comment]
@t-hamano Thanks for the response.

Are you referring to `layout` coming from: `"supports": { "layout": { } }` or a totally custom `layout` attribute for which I will need to write custom code to style the layout?

**Comment by t-hamano** [comment]
> Are you referring to `layout` coming from: `"supports": { "layout": { } }` or a totally custom `layout` attribute for which I will need to write custom code to style the layout?

I mean the layout attribute as a _custom_ attribute, not `supports.layout`. From your screencast, it looks like you want to change the order of the blocks inside when you change the layout. However, `supports.layout` has no such functionality.

I don't think the nested group block is necessary if you want to change the type of flex layout or change the order of blocks inside depending on the layout settings of the custom block.

**Comment by danieldudzic** [comment]
> I don't think the nested group block is necessary if you want to change the type of flex layout or change the order of blocks inside depending on the layout settings of the custom block.

Right, it's not necessary, in terms of creating a custom solution for it, but it's necessary if I'd like to utilize the Core layout controls, which is what I'm trying to do :)

Here's some more context: https://github.com/woocommerce/woocommerce-blocks/discussions/10326
